### ISSUE TITLE
libmoosex-app-perl: FTBFS: dh_auto_test: error: make -j4 test TEST_VERBOSE=1 returned exit code 2

### ISSUE THREAD

**Issue by gregoa** [maros/MooseX-App#61]
We have the following bug reported to the Debian package of
MooseX-App, c.f. https://bugs.debian.org/978302

It doesn't seem to be a bug in the packaging, so you may want to take
a look. Thanks!

This test failure is quite new, my guess is that it might be related to the
recent update of Moose to 2.2014.

------8<-----------8<-----------8<-----------8<-----------8<-----

```
Source: libmoosex-app-perl
Version: 1.41-1
Severity: serious
Justification: FTBFS on amd64
Tags: bullseye sid ftbfs
Usertags: ftbfs-20201226 ftbfs-bullseye

Hi,

During a rebuild of all packages in sid, your package failed to build
on amd64.

Relevant part (hopefully):
> make[1]: Entering directory '/<<PKGBUILDDIR>>'
> PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(1, 'blib/lib', 'blib/arch')" t/*.t
> t/00_load.t ............... 
> 1..43
> ok 1 - use MooseX::App;
> ok 2 - use MooseX::App::ParsedArgv;
> ok 3 - use MooseX::App::Command;
> ok 4 - use MooseX::App::Utils;
> ok 5 - use MooseX::App::Message::Block;
> ok 6 - use MooseX::App::Message::Envelope;
> ok 7 - use MooseX::App::Meta::Role::Attribute::Option;
> ok 8 - use MooseX::App::Meta::Role::Class::Base;
> ok 9 - use MooseX::App::Meta::Role::Class::Command;
> ok 10 - use MooseX::App::Meta::Role::Class::Simple;
> ok 11 - use MooseX::App::Meta::Role::Class::Documentation;
> ok 12 - use MooseX::App::Plugin::BashCompletion;
> ok 13 - use MooseX::App::Plugin::BashCompletion::Command;
> ok 14 - use MooseX::App::Plugin::BashCompletion::Meta::Class;
> ok 15 - use MooseX::App::Plugin::Config;
> ok 16 - use MooseX::App::Plugin::Config::Meta::Class;
> ok 17 - use MooseX::App::Plugin::Version;
> ok 18 - use MooseX::App::Plugin::Version::Command;
> ok 19 - use MooseX::App::Plugin::Version::Meta::Class;
> ok 20 - use MooseX::App::Plugin::Man;
> ok 21 - use MooseX::App::Plugin::Man::Command;
> ok 22 - use MooseX::App::Plugin::Man::Meta::Class;
> ok 23 - use MooseX::App::Plugin::Depends;
> ok 24 - use MooseX::App::Plugin::Depends::Meta::Attribute;
> ok 25 - use MooseX::App::Plugin::Depends::Meta::Class;
> ok 26 - use MooseX::App::Plugin::MutexGroup;
> ok 27 - use MooseX::App::Plugin::MutexGroup::Meta::Attribute;
> ok 28 - use MooseX::App::Plugin::MutexGroup::Meta::Class;
> ok 29 - use MooseX::App::Utils;
> ok 30 - use MooseX::App::Simple;
> ok 31 - use MooseX::App::Exporter;
> ok 32 - use MooseX::App::Role::Base;
> ok 33 - use MooseX::App::Role::Common;
> ok 34 - use MooseX::App::Plugin::Term;
> ok 35 - use MooseX::App::Plugin::Term::Meta::Class;
> ok 36 - use MooseX::App::Plugin::Term::Meta::Attribute;
> ok 37 - use MooseX::App::Plugin::Color;
> ok 38 - use MooseX::App::Message::BlockColor;
> ok 39 - use MooseX::App::Plugin::Color::Meta::Class;
> ok 40 - use MooseX::App::Plugin::ConfigHome;
> ok 41 - use MooseX::App::Plugin::ConfigHome::Meta::Class;
> ok 42 # skip Text::WagnerFischer is not installed
> ok 43 # skip Text::WagnerFischer is not installed
> ok
> t/01_basic.t .............. 
> 1..13
> # Subtest: Excact command with option
>     ok 1 - An object of class 'Test01::CommandA' isa 'Test01::CommandA'
>     ok 2 - Param is set
>     1..2
> ok 1 - Excact command with option
> # Subtest: Fuzzy command with option
>     ok 1 - An object of class 'Test01::CommandA' isa 'Test01::CommandA'
>     ok 2 - Param is set
>     1..2
> ok 2 - Fuzzy command with option
> # Subtest: Wrong command
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Message is set
>     ok 3 - Message is of type error
>     ok 4 - Usage set
>     ok 5 - Usage body set
>     ok 6 - Global options set
>     ok 7 - Global options body set
>     ok 8 - Available commands set
>     ok 9 - Available commands body set
>     1..9
> ok 3 - Wrong command
> # Subtest: Help for command
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Usage is set
>     ok 3 - Usage body set
>     ok 4 - Description is set
>     ok 5 - Description body set
>     ok 6 - Options header is set
>     ok 7 - Options body is set
>     1..7
> ok 4 - Help for command
> # Subtest: With extra args
>     ok 1 - An object of class 'Test01::CommandB' isa 'Test01::CommandB'
>     ok 2 - Param global is set
>     ok 3 - Param param_b is set
>     ok 4 - Param private is set
>     1..4
> ok 5 - With extra args
> # Subtest: Wrapper script
>     ok 1 - Output is ok
>     1..1
> ok 6 - Wrapper script
> # Subtest: Custom help text
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Usage is set
>     ok 3 - Usage is ok
>     ok 4 - description is set
>     ok 5 - Description is ok
>     1..5
> ok 7 - Custom help text
> # Subtest: Input errors missing
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Error message ok
>     1..2
> ok 8 - Input errors missing
> # Subtest: Input errors type
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Error message ok
>     ok 3 - Error message ok
>     1..3
> ok 9 - Input errors type
> # Subtest: Global help requested
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Help message ok
>     1..2
> ok 10 - Global help requested
> # Subtest: Missing command
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Error message ok
>     1..2
> ok 11 - Missing command
> # Subtest: Extra params
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Error message ok
>     1..2
> ok 12 - Extra params
> ok 13 - no warnings
> ok
> t/02_meta.t ............... 
> 1..29
> ok 1 - Commands found
> ok 2 - Commands found
> ok 3 - Command namespace ok
> ok 4 - Command base ok
> ok 5 - Message class
> ok 6 - Role applied to base class
> ok 7 - Role applied to base class
> ok 8 - Found three commands
> ok 9 - Command A found
> ok 10 - Command A matched
> ok 11 - Command A,B,C1 and D matched
> ok 12 - Command C1 matched fuzzy
> ok 13 - Command C1 matched exactly
> ok 14 - Has three attributes
> ok 15 - Usage name ok
> ok 16 - Usage description ok
> ok 17 - Usage name ok
> ok 18 - Usage name ok
> ok 19 - Usage description ok
> ok 20 - Tags ok
> ok 21 - Changed tags ok
> ok 22 - An object of class 'MooseX::App::Message::Block' isa 'MooseX::App::Message::Block'
> ok 23 - Description body ok
> ok 24 - Pod short description parsed ok
> ok 25 - Pod long description parsed ok
> ok 26 - Command usage parsed ok
> ok 27 - Attribute name ok
> ok 28 - Check attributes tags
> ok 29 - no warnings
> ok
> t/03_utils.t .............. 
> 1..5
> # Subtest: Class to command
>     ok 1 - Command ok
>     ok 2 - Command ok
>     ok 3 - Command ok
>     ok 4 - Command ok
>     ok 5 - Command ok
>     1..5
> ok 1 - Class to command
> # Subtest: Format text
>     ok 1 - Format text ok
>     ok 2 - Format text ok
>     ok 3 - Format text ok
>     1..3
> ok 2 - Format text
> # Subtest: Formater
>     ok 1 - Format list ok
>     1..1
> ok 3 - Formater
> # Subtest: Parser
>     ok 1 - Has 13 elements
>     ok 2 - Parameter parsed ok
>     ok 3 - Parameter type ok
>     ok 4 - Flag parsed ok
>     ok 5 - Flag type ok
>     ok 6 - Parameter parsed ok
>     ok 7 - Flag parsed ok
>     ok 8 - Flag parsed ok
>     ok 9 - Flag parsed ok
>     ok 10 - Flag parsed ok
>     ok 11 - Option parsed ok
>     ok 12 - Option value ok
>     ok 13 - Parameter parsed ok
>     ok 14 - Parameter parsed ok
>     ok 15 - Option value ok
>     ok 16 - Option parsed ok
>     ok 17 - Extra parsed ok
>     ok 18 - Option value ok
>     ok 19 - Extra parsed ok
>     ok 20 - Extra parsed ok
>     1..20
> ok 4 - Parser
> ok 5 - no warnings
> ok
> t/04_plugin_config.t ...... 
> 1..5
> # Subtest: Command with config
>     ok 1 - An object of class 'Test01::CommandA' isa 'Test01::CommandA'
>     ok 2 - Arg from command config
>     ok 3 - Arg from command config
>     ok 4 - Config file is set
>     ok 5 - Config loaded
>     1..5
> ok 1 - Command with config
> # Subtest: Another command with config
>     ok 1 - An object of class 'Test01::CommandB' isa 'Test01::CommandB'
>     ok 2 - Arg from command config
>     1..2
> ok 2 - Another command with config
> # Subtest: Command with config and argv
>     ok 1 - An object of class 'Test01::CommandA' isa 'Test01::CommandA'
>     ok 2 - Arg from command config
>     ok 3 - Arg from command config
>     1..3
> ok 3 - Command with config and argv
> # Subtest: Missing config
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Error message set
>     1..2
> ok 4 - Missing config
> ok 5 - no warnings
> ok
> 
>     #   Failed test 'Check enum error message'
>     #   at t/05_extended.t line 244.
>     #          got: 'Validation failed for '__ANON__' with value ggg. Value must be equal to "aaa", "bbb", "ccc", "ddd", "eee", or "fff".'
>     #     expected: 'Value must be one of these values: aaa, bbb, ccc, ddd, eee, fff (not 'ggg')'
>     # Looks like you failed 1 test of 2.
> 
> #   Failed test 'Test enum error message'
> #   at t/05_extended.t line 245.
> # Looks like you failed 1 test of 30.
> t/05_extended.t ........... 
> 1..30
> # Subtest: Non-Fuzzy command matching
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Message ok
>     1..2
> ok 1 - Non-Fuzzy command matching
> # Subtest: Non-Fuzzy attribute matching
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Message ok
>     1..2
> ok 2 - Non-Fuzzy attribute matching
> # Subtest: Private option is not exposed
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Message ok
>     ok 3 - Message is of type error
>     1..3
> ok 3 - Private option is not exposed
> # Subtest: Options from role
>     ok 1 - An object of class 'Test03::SomeCommand' isa 'Test03::SomeCommand'
>     ok 2 - Param is set
>     ok 3 - Role param is set
>     1..3
> ok 4 - Options from role
> # Subtest: Missing attribute value
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Message ok
>     ok 3 - Message is of type error
>     1..3
> ok 5 - Missing attribute value
> # Subtest: All options available & no description
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - No description
>     ok 3 - Message ok
>     1..3
> ok 6 - All options available & no description
> # Subtest: Test type constraints integer
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Message ok
>     1..2
> ok 7 - Test type constraints integer
> # Subtest: Test type constraints hash
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Message ok
>     1..2
> ok 8 - Test type constraints hash
> # Subtest: Test type constraints number
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Message ok
>     1..2
> ok 9 - Test type constraints number
> # Subtest: Test type constraints custom1
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Message ok
>     1..2
> ok 10 - Test type constraints custom1
> # Subtest: Test pass type constraints
>     ok 1 - An object of class 'Test03::AnotherCommand' isa 'Test03::AnotherCommand'
>     ok 2 - Hash ok
>     ok 3 - Hash ok
>     ok 4 - Integer ok
>     ok 5 - Custom type 1 ok
>     ok 6 - Custom type 2 ok
>     ok 7 - Custom type 2 ok
>     ok 8 - Attr set ok
>     ok 9 - Count set ok
>     ok 10 - Split ok
>     1..10
> ok 11 - Test pass type constraints
> # Subtest: Test ambiguous options
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Message ok
>     ok 3 - Message ok
>     1..3
> ok 12 - Test ambiguous options
> # Subtest: Test flags & defaults
>     ok 1 - An object of class 'Test03::YetAnotherCommand' isa 'Test03::YetAnotherCommand'
>     ok 2 - Bool1 flag is undef
>     ok 3 - Bool2 flag is set
>     ok 4 - Bool3 flag is set
>     ok 5 - Value is default
>     1..5
> ok 13 - Test flags & defaults
> # Subtest: Test more flags & defaults
>     ok 1 - An object of class 'Test03::YetAnotherCommand' isa 'Test03::YetAnotherCommand'
>     ok 2 - Bool1 flag is undef
>     ok 3 - Bool2 flag is unset
>     ok 4 - Bool3 flag is set
>     ok 5 - Value is set
>     1..5
> ok 14 - Test more flags & defaults
> # Subtest: Test Negate boolean options
>     ok 1 - An object of class 'Test03::YetAnotherCommand' isa 'Test03::YetAnotherCommand'
>     ok 2 - Bool3 flag is negated
>     1..2
> ok 15 - Test Negate boolean options
> # Subtest: Test positional params
>     ok 1 - An object of class 'Test03::ExtraCommand' isa 'Test03::ExtraCommand'
>     ok 2 - Extra1 value is "hui"
>     ok 3 - Extra2 value is undef
>     ok 4 - alpha value is undef
>     ok 5 - Value is set
>     1..5
> ok 16 - Test positional params
> # Subtest: Test positional params
>     ok 1 - An object of class 'Test03::ExtraCommand' isa 'Test03::ExtraCommand'
>     ok 2 - Extra1 value is "hui"
>     ok 3 - Extra2 value is undef
>     ok 4 - alpha value is undef
>     ok 5 - Value is set
>     1..5
> ok 17 - Test positional params
> # Subtest: Test optional positional params
>     ok 1 - An object of class 'Test03::ExtraCommand' isa 'Test03::ExtraCommand'
>     ok 2 - Extra1 value is "hui"
>     ok 3 - Extra2 value is "11"
>     ok 4 - alpha value is undef
>     ok 5 - Value is set
>     1..5
> ok 18 - Test optional positional params
> # Subtest: Test wrong positional params
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Error message ok
>     ok 3 - Usage header ok
>     ok 4 - Usage body ok
>     1..4
> ok 19 - Test wrong positional params
> # Subtest: Test missing positional params
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Message ok
>     1..2
> ok 20 - Test missing positional params
> # Subtest: Test extra positional params
>     ok 1 - An object of class 'Test03::ExtraCommand' isa 'Test03::ExtraCommand'
>     ok 2 - Param 1 ok
>     ok 3 - Param 2 ok
>     ok 4 - Param 3 ok
>     ok 5 - Uncomsumed option ok
>     1..5
> ok 21 - Test extra positional params
> # Subtest: Test parameter preference
>     ok 1 - An object of class 'Test03::ExtraCommand' isa 'Test03::ExtraCommand'
>     ok 2 - Extra param from new_with_command ok
>     ok 3 - Extra param from argv ok
>     ok 4 - value option from argv ok
>     ok 5 - Flag from new_with_command ok
>     ok 6 - Flago from new_with_command ok
>     1..6
> ok 22 - Test parameter preference
> # Subtest: Test parameter preference reverse
>     ok 1 - An object of class 'Test03::ExtraCommand' isa 'Test03::ExtraCommand'
>     ok 2 - Extra param from new_with_command ok
>     ok 3 - Extra param from argv ok
>     ok 4 - value option from argv ok
>     ok 5 - Flag from new_with_command ok
>     ok 6 - Flago from new_with_command ok
>     1..6
> ok 23 - Test parameter preference reverse
> # Subtest: Test enum error message
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     not ok 2 - Check enum error message
>     1..2
> not ok 24 - Test enum error message
> # Subtest: Test empty multi
>     ok 1 - An object of class 'Test03::SomeCommand' isa 'Test03::SomeCommand'
>     ok 2 - Has three list items
>     ok 3 - First value ok
>     ok 4 - Second value ok
>     ok 5 - First value empty
>     1..5
> ok 25 - Test empty multi
> # Subtest: Test permute
>     ok 1 - Permute array ok
>     ok 2 - Permute hash ok
>     1..2
> ok 26 - Test permute
> # Subtest: Test permute off
>     ok 1 - No permute array ok
>     ok 2 - No permute hash ok
>     1..2
> ok 27 - Test permute off
> # Subtest: Test parameter order
>     ok 1 - Param a ok
>     ok 2 - Param b ok
>     ok 3 - Param c ok
>     1..3
> ok 28 - Test parameter order
> # Subtest: Test mixed multi
>     ok 1 - An object of class 'Test03::SomeCommand' isa 'Test03::SomeCommand'
>     ok 2 - List ok
>     1..2
> ok 29 - Test mixed multi
> ok 30 - no warnings
> Dubious, test returned 1 (wstat 256, 0x100)
> Failed 1/30 subtests 
> t/06_env.t ................ 
> 1..6
> # Subtest: Command with argv
>     ok 1 - An object of class 'Test01::CommandA' isa 'Test01::CommandA'
>     ok 2 - Arg from command config
>     1..2
> ok 1 - Command with argv
> # Subtest: Command only with env
>     ok 1 - An object of class 'Test01::CommandA' isa 'Test01::CommandA'
>     ok 2 - Arg from command env
>     1..2
> ok 2 - Command only with env
> # Subtest: Command with env and argv
>     ok 1 - An object of class 'Test01::CommandA' isa 'Test01::CommandA'
>     ok 2 - Arg from command argv
>     1..2
> ok 3 - Command with env and argv
> # Subtest: Env not passing type constraint
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Message ok
>     1..2
> ok 4 - Env not passing type constraint
> # Subtest: Coerce env value
>     ok 1 - An object of class 'Test01::CommandD' isa 'Test01::CommandD'
>     ok 2 - Arg coerced from command env
>     1..2
> ok 5 - Coerce env value
> ok 6 - no warnings
> ok
> t/07_single.t ............. 
> 1..4
> # Subtest: Single command
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Check for error message
>     1..2
> ok 1 - Single command
> # Subtest: Single command
>     ok 1 - An object of class 'Test05' isa 'Test05'
>     ok 2 - Arg from command ARGV
>     ok 3 - Arg from new_with_options
>     1..3
> ok 2 - Single command
> # Subtest: Single command help
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Usage header is first
>     1..2
> ok 3 - Single command help
> ok 4 - no warnings
> ok
> t/08_plugin_various.t ..... 
> 1..3
> # Subtest: Bash completion
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - some_command present
>     ok 3 - global_option present
>     ok 4 - roleattr present
>     ok 5 - bash_completion is not included
>     1..5
> ok 1 - Bash completion
> # Subtest: Version
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Check for app version
>     ok 3 - Check for MooseX::App version
>     ok 4 - License included
>     1..4
> ok 2 - Version
> ok 3 - no warnings
> ok
> t/09_classes.t ............ 
> 1..7
> # Subtest: Extend base class
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - --test1 included
>     ok 3 - --test2 included
>     ok 4 - --test3 not included
>     1..4
> ok 1 - Extend base class
> # Subtest: Wrong usage
>     ok 1 - Only callable as class method
>     ok 2 - new_with_command may only be called from the application base package
>     ok 3 - Wrong default args
>     1..3
> ok 2 - Wrong usage
> # Subtest: Conflicts
>     ok 1 - Conflict detected
>     1..1
> ok 3 - Conflicts
> # Subtest: Default args available with extra inheritance
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Help ok
>     ok 3 - Help ok
>     ok 4 - An object of class 'Test03::YetAnotherCommand' isa 'Test03::YetAnotherCommand'
>     ok 5 - Option has been passed on
>     1..5
> ok 4 - Default args available with extra inheritance
> # Subtest: Attributes from role
>     ok 1 - An object of class 'Test03::SomeCommand' isa 'Test03::SomeCommand'
>     ok 2 - Attribute from role ok
>     1..2
> ok 5 - Attributes from role
> # Subtest: Correct order from role 
>     ok 1 - An object of class 'Test03::SomeCommand' isa 'Test03::SomeCommand'
>     ok 2 - First from role
>     ok 3 - Second from role
>     ok 4 - Third from role
>     1..4
> ok 6 - Correct order from role 
> ok 7 - no warnings
> ok
> t/10_plugin_mutexgroup.t .. 
> 1..2
> # Subtest: MutexGroup
>     1..7
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - only returned a single error
>     ok 3 - generated an error when more than one option in the same mutexgroup is initialized
>     ok 4 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 5 - only returned a single error
>     ok 6 - generated an error when no options in the same mutexgroup are initialized
>     ok 7 - generated no errors when only a single option from the same mutexgroup is initialized
> ok 1 - MutexGroup
> ok 2 - no warnings
> ok
> t/11_process.t ............ 
> 1..2
> # Subtest: Test basic exit codes
>     ok 1 - Exitcode test02.pl ok
>     ok 2 - Output test02.pl ok
>     ok 3 - Error test02.pl ok
>     ok 4 - Exitcode test02.pl error ok
>     ok 5 - Error test02.pl error ok
>     ok 6 - Exitcode test02.pl version ok
>     ok 7 - Output test02.pl version ok
>     ok 8 - Error test02.pl version ok
>     ok 9 - Exitcode test02.pl version ok
>     ok 10 - Output test02.pl version ok
>     ok 11 - Error test02.pl version ok
>     ok 12 - Exitcode test02.pl record --help ok
>     ok 13 - Output test02.pl record --help ok
>     ok 14 - Error test02.pl record --help ok
>     ok 15 - Exitcode test02.pl record ok
>     ok 16 - Output test02.pl record ok
>     ok 17 - Error test02.pl record ok
>     ok 18 - Exitcode test02.pl record --notthere ok
>     ok 19 - Output test02.pl record --notthere ok
>     ok 20 - Error test02.pl record --notthere ok
>     1..20
> ok 1 - Test basic exit codes
> ok 2 - no warnings
> ok
> t/12_plugin_depends.t ..... 
> 1..2
> # Subtest: Depends
>     1..8
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - only returned a single error
>     ok 3 - generated an error when an option dependency was not present
>     ok 4 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 5 - only returned a single error
>     ok 6 - generated an error when an option dependency was not present
>     ok 7 - generated no errors when both an option and its dependencies are defined
>     ok 8 - generated no errors when both an option and its dependencies are defined
> ok 1 - Depends
> ok 2 - no warnings
> ok
> t/13_rt_112156.t .......... 
> 1..3
> # Subtest: no inheritance
>     1..8
>     ok 1 - An object of class 'Test13::SomeCommand' isa 'Test13::SomeCommand'
>     ok 2 - Option ok
>     ok 3 - Option ok
>     ok 4 - No option
>     ok 5 - An object of class 'Test13::AnotherCommand' isa 'Test13::AnotherCommand'
>     ok 6 - Option ok
>     ok 7 - Option ok
>     ok 8 - Option ok
> ok 1 - no inheritance
> # Subtest: check plugin functionality
>     1..2
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
> ok 2 - check plugin functionality
> ok 3 - no warnings
> ok
> t/14_gh_45.t .............. 
> 1..4
> # Subtest: boolean negation
>     1..4
>     ok 1 - An object of class 'Test14' isa 'Test14'
>     ok 2 - Did not set b1
>     ok 3 - Did set not_b1
>     ok 4 - Did set b2
> ok 1 - boolean negation
> # Subtest: boolean fuzzy negation
>     1..4
>     ok 1 - An object of class 'Test14' isa 'Test14'
>     ok 2 - Did not set b1
>     ok 3 - Did set not_b1
>     ok 4 - Did set b2
> ok 2 - boolean fuzzy negation
> # Subtest: ambiguous negation
>     1..3
>     ok 1 - An object of class 'Test14' isa 'Test14'
>     ok 2 - Did  set b1
>     ok 3 - Did not set b2
> ok 3 - ambiguous negation
> ok 4 - no warnings
> ok
> t/15_subcommands.t ........ 
> 1..4
> # Subtest: Basic Subcommands
>     ok 1 - An object of class 'Test15::Foo' isa 'Test15::Foo'
>     ok 2 - An object of class 'Test15::Foo::Bar' isa 'Test15::Foo::Bar'
>     ok 3 - An object of class 'Test15::Foo::Baz' isa 'Test15::Foo::Baz'
>     ok 4 - An object of class 'Test15::Foo' isa 'Test15::Foo'
>     1..4
> ok 1 - Basic Subcommands
> # Subtest: Help Subcommand
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Command headline set
>     ok 3 - Command body set
>     1..3
> ok 2 - Help Subcommand
> # Subtest: Help Parent
>     ok 1 - An object of class 'MooseX::App::Message::Envelope' isa 'MooseX::App::Message::Envelope'
>     ok 2 - Command headline set
>     ok 3 - Command body set
>     1..3
> ok 3 - Help Parent
> ok 4 - no warnings
> ok
> t/16_rt122946.t ........... 
> 1..2
> # Subtest: Option flags
>     ok 1 - threw Regexp ((?^:Option param[ab] has a single letter flag but no Bool))
>     1..1
> ok 1 - Option flags
> ok 2 - no warnings
> ok
> 
> Test Summary Report
> -------------------
> t/05_extended.t         (Wstat: 256 Tests: 30 Failed: 1)
>   Failed test:  24
>   Non-zero exit status: 1
> Files=17, Tests=164, 11 wallclock secs ( 0.06 usr  0.03 sys + 10.40 cusr  0.71 csys = 11.20 CPU)
> Result: FAIL
> Failed 1/17 test programs. 1/164 subtests failed.
> make[1]: *** [Makefile:973: test_dynamic] Error 255
> make[1]: Leaving directory '/<<PKGBUILDDIR>>'
> dh_auto_test: error: make -j4 test TEST_VERBOSE=1 returned exit code 2

The full build log is available from:
   http://qa-logs.debian.net/2020/12/26/libmoosex-app-perl_1.41-1_unstable.log

A list of current common problems and possible solutions is available at
http://wiki.debian.org/qa.debian.org/FTBFS . You're welcome to contribute!

If you reassign this bug to another package, please marking it as 'affects'-ing
this package. See https://www.debian.org/Bugs/server-control#affects

If you fail to reproduce this, please provide a build log and diff it with me
so that we can identify if something relevant changed in the meantime.

About the archive rebuild: The rebuild was done on EC2 VM instances from
Amazon Web Services, using a clean, minimal and up-to-date chroot. Every
failed build was retried once to eliminate random failures.


```
------8<-----------8<-----------8<-----------8<-----------8<-----

Thanks for considering,
  gregor herrmann,
  Debian Perl Group

**Comment by soig** [comment]
It's not related to "make test -j", the "-j" is unrelated to the failure (which happens on recent Mageia Cauldron too).
It's likely because of another perl module upgrade (Moose).

See autobuild history for perl-MooseX-App on Mageia Linux:
https://pkgsubmit.mageia.org/autobuild/history.php?package=perl-MooseX-App

Compare last working with first not working:
https://pkgsubmit.mageia.org/autobuild/cauldron/aarch64/core/2020-12-26/perl-MooseX-App-1.410.0-2.mga8.src.rpm/rpm_qa.0.20201226025645.log
https://pkgsubmit.mageia.org/autobuild/cauldron/aarch64/core/2020-12-18/perl-MooseX-App-1.410.0-2.mga8.src.rpm/rpm_qa.0.20201218120516.log

It started failing after updating Moose from 2.201.300 to 2.201.400

**Comment by soig** [comment]
Removing Devel-PartialDump slightly alters the message:
-    #          got: 'Validation failed for '__ANON__' with value "ggg". Value must be equal to "aaa", "bbb", "ccc", "ddd", "eee", or "fff".'
+    #          got: 'Validation failed for '__ANON__' with value ggg. Value must be equal to "aaa", "bbb", "ccc", "ddd", "eee", or "fff".'
Likely because of https://github.com/moose/Moose/commit/86ecb88a13b4c47cfa5f01994e71c2bf76553d96 & parent commits

The attached patches fix running testsuite with latest Moose.

[fix-test-with-Moose-2.201.400.patch.txt](https://github.com/maros/MooseX-App/files/5764223/fix-test-with-Moose-2.201.400.patch.txt)

**Comment by maros** [comment]
Duplicate of #62 . will release fix soon. Thank you
